lib/rucio/common/dumper/consistency.py

```diff
@@ -301,7 +304,8 @@ def gnu_sort(file_path, prefix=None, delimiter=None, fieldspec=None, cache_dir=D
     memory and it is relatively fast if used with the environment variable
     LC_ALL set to C as in this function.
     '''
-    assert (delimiter is None and fieldspec is None) or (delimiter is not None and fieldspec is not None)
+    if (delimiter is None and fieldspec is not None) or (delimiter is not None and fieldspec is None):
```

Comment:
This is `XOR(delimiter, fieldspec)`, so it could be simplified either like this:

```suggestion
    if (delimiter is not None) != (fieldspec is not None):
```

or like this:

```suggestion
    if (delimiter is not None) ^ (fieldspec is not None):
```

I think the first one is clearer in this context, especially cause the `ValueError` message perfectly describes the issue, but the second one is in theory the correct way to write a XOR operation

See also: https://stackoverflow.com/questions/432842/how-do-you-get-the-logical-xor-of-two-variables-in-python

Reply:
I have never seen that ^ notation for XOR, thanks for teaching me something new and fun